pydantic_settings/sources.py

```diff
@@ -126,6 +126,36 @@ class PydanticBaseSettingsSource(ABC):
     def __init__(self, settings_cls: type[BaseSettings]):
         self.settings_cls = settings_cls
         self.config = settings_cls.model_config
+        self._previous_state: dict[str, Any] = {}
+        self._previous_states: dict[str, dict[str, Any]] = {}
+
+    def _set_previous_state(self, state: dict[str, Any]) -> None:
+        """
+        Record the state of settings from the previous settings sources. This should
+        be called right before __call__.
+        """
+        self._previous_state = state
+
+    def _set_previous_states(self, states: dict[str, dict[str, Any]]) -> None:
+        """
+        Record the state of settings from all previous settings sources. This should
+        be called right before __call__.
+        """
+        self._previous_states = states
+
+    @property
+    def previous_states(self) -> dict[str, dict[str, Any]]:
```

Comment:
I think `previous_states` and `previous_state` may make user confused. what about replacing  `previous_states` with `settings_sources_data` and `previous_state` with `current_state`?

If you have any suggestions please let me know.

Reply:
I have `current_settings_state` in mind, but it's uselessly verbose.

I don't find `settings_sources_data` particularly explicit, but I cannot find a better name. Maybe `previous_settings_sources_output`, or `previous_sources_output` (a little shorter)?

In any case, I've implemented your suggestions. Let me know what you think!